lib/ecto/migration/dsl.ex

```diff
@@ -0,0 +1,297 @@
+  Modify a column when altering a table.
+
+  ## Examples
+
+      alter table(:posts) do
+        modify :title, :text
+      end
+
+  """
+  def modify(column, type, opts \\ []) do
+    {:modify, column, type, opts}
+  end
+
+  @doc """
+  Remove a column when altering a table.
+
+  ## Examples
+
+      alter table(:posts) do
+        remove :title
+      end
+
+  """
+  def remove(column) do
+    {:remove, column}
+  end
+
+  @doc """
+  Rename a column when altering a table.
+
+  ## Examples
+
+      alter table(:posts) do
+        rename :name, :title
+      end
+
+  """
+  def rename(from, to) do
+    {:rename, from, to}
+  end
+
+  @doc """
+  Add a single column. Shortcut for using `alter/1`.
+
+  ## Examples
+
+      add_column :products, :summary, :string
+
+  """
+  def add_column(table_name, column, type, options \\ []) do
+    alter table(table_name) do
+      add(column, type, options)
+    end
+  end
+
+  @doc """
+  Modify a single column's type. Shortcut for using `alter/1`.
+
+  ## Examples
+
+      modify_column(:user, :rating, :integer)
+
+  """
+  def modify_column(table_name, column, type, options \\ []) do
+    alter table(table_name) do
+      modify(column, type, options)
+    end
+  end
+
+  @doc """
+  Remove a single column. Shortcut for using `alter/1`.
+
+  ## Examples
+
+      remove_column :products, :title
+
+  """
+  def remove_column(table_name, column) do
+    alter table(table_name) do
+      remove(column)
+    end
+  end
+
+  @doc """
+  Rename a single column. Shortcut for using `alter/1`.
+
+  ## Examples
+
+      rename_column :products, :old_name, :new_name
+
+  """
+  def rename_column(table_name, from, to) do
+    alter table(table_name) do
+      rename(from, to)
+    end
+  end
+
+  @doc """
+  Adds `created_at` and `updated_at` columns to a table.
+
+  ## Examples
+
+      create table(:posts) do
+        timestamps
+      end
+
+  """
+  def timestamps do
+    [add(:created_at, :datetime), add(:updated_at, :datetime)]
+  end
+
+  @doc """
+  Adds a foreign key.
+
+  ## Examples
+
+      create table(:product) do
+        add :category_id, references(:category)
+      end
+
+  ## Options
+
+  * `:foreign_column` The foreign column's name, default is `:id`
+  * `:type`           The foreign column's type, default is `:integer`
+
+  """
+  def references(table, opts \\ []) do
+    foreign_column = Keyword.get(opts, :foreign_column, :id)
+    type           = Keyword.get(opts, :type, :integer)
+
+    {:references, table, foreign_column, type}
+  end
+
+  @doc """
+  Checks if a column exists.
+
+  ## Examples
+
+    if !column_exists?(:products, :name) do
+      add_column(:products, :name, :string)
+    end
+
+  """
+  def column_exists?(table_name, column_name) do
+    exists?(:column, {table_name, column_name})
+  end
+
+  @doc """
+  Checks if a table exists.
+
+  ## Examples
+
+    if table_exists?(:products) do
+      drop table(:products)
+    end
+
+  """
+  def table_exists?(table_name) do
```

Comment:
It could be... my concern is if people would then try to run something like `drop column(:products, :name)` instead of using remove. :S

Maybe we could allow `exists?(:column)` but only inside an create/alter table?